static/app/views/prevent/tests/queries/useRepo.tsx

```diff
@@ -0,0 +1,28 @@
+import {useApiQuery} from 'sentry/utils/queryClient';
+import type {UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+
+export interface Repository {
+  testAnalyticsEnabled: boolean;
+  uploadToken?: string;
+}
+
+export function useRepo({
+  organizationSlug,
+  integratedOrgId,
+  repository,
+}: {
+  integratedOrgId?: string;
+  organizationSlug?: string;
+  repository?: string;
```

Comment:
Thoughts on calling useOrganization and usePreventContext here instead of passing these as props?

If you need these for the URL, then I think it's fair to have them belong to the hook itself. Also, since you need them for the call, I don't think they should be optional either